00_integrity_checker.R

```diff
@@ -144,138 +123,106 @@ check_data_types <- function(barefile, quotedfile) {
     
     y <- cols_and_data_types %>% 
       left_join(data_types, by = c("File", "Column")) %>%
+      filter(DataType != ImportedDataType) %>%
       mutate(
-        Issue = if_else(DataType != ImportedDataType, 
-                        "Incorrect Data Type",
-                        "nothing"), 
-        Guidance = if_else(
-          DataType != ImportedDataType,
-          str_squish(paste0(
-            "In the ",
-            quotedfile,
-            " file, the ",
-            Column,
-            " column should have a data type of ",
-            case_when(
-              DataType == "numeric" ~ "integer",
-              DataType == "character" ~ "string",
-              TRUE ~ DataType
+        Issue = "Incorrect Data Type",
+        Type = if_else(DataTypeHighPriority == 1, "High Priority", "Error"),
+        Guidance = str_squish("Data types must align with the HMIS CSV Format Specifications. 
+                    Please review the Specifications for the data types associated 
+                    with the file and column listed in the detail and make the necessary updates."),
+        Detail = str_squish(paste0(
+          "In the ",
+          quotedfile,
+          " file, the ",
+          Column,
+          " column should have a data type of ",
+          case_when(
+            DataType == "numeric" ~ "integer",
+            DataType == "character" ~ "string",
+            TRUE ~ DataType
+          ),
+          " but in this file, it is",
+          case_when(
+            ImportedDataType == "numeric" ~ "integer",
+            ImportedDataType == "character" ~ "string",
+            TRUE ~ ImportedDataType
             ),
-            " but in this file, it is",
-            case_when(
-              ImportedDataType == "numeric" ~ "integer",
-              ImportedDataType == "character" ~ "string",
-              TRUE ~ ImportedDataType
-              ),
-            ". The PersonalID must be unique within the Client.csv."
-          )),
-          NULL
-        ),
-        Type = if_else(DataTypeHighPriority == 1, "High Priority", "Error")) %>%
-      filter(Issue != "nothing") %>%
-      select(Issue, Type, Guidance)
-    
+          ". The PersonalID must be unique within the Client.csv."
```

Comment:
This line seems like it's maybe out of place unless the logic was changed so that we're checking whether the PersonalID is unique together with the data type logic

Reply:
this change was from commit e9210064. However, you can see from the diff that this line wasn't new; it was already in there from commit 85766d2 (dated 12/12/22). I just don't want to drop something that we wanted in there for a reason.